File: PortalDietetyczny.Client/src/app/DTOs/RecipeDetailsDto.ts
import { NutritionInfo, RecipeIngredientDto } from "./AddRecipeDto";

export interface RecipeDetailsDto
{
  uid: number;
  recipeTags: string[];
  name: string;
  nutrition: NutritionInfo;
  ingredients: RecipeIngredientDto[];
  instruction: string;
  photoUrl?: string;
}

File: PortalDietetycznyAPI/Domain/DTOs/RecipeDetailsDto.cs
using PortalDietetycznyAPI.Domain.Entities;

namespace PortalDietetycznyAPI.DTOs;

public class RecipeDetailsDto
{
    public int Uid { get; set; }
    public List<string> RecipeTags { get; set; } = [];
    public string Name { get; set; }
    public NutritionInfo Nutrition { get; set; } = new NutritionInfo();
    public List<RecipeIngredientDto> Ingredients { get; set; } = [];
    public string Instruction { get; set; }
    public string? PhotoUrl { get; set; }
}